code/datums/ship.dm

```diff
@@ -97,6 +101,8 @@ GLOBAL_VAR(next_ship_id)
 		C.x_loc = text2num(coords[1])
 		C.y_loc = text2num(coords[2])
 		C.ship = src
+		if(C.attack_data)
+			C.attack_data.ship = src
```

Comment:
Moving this into component would require adding a callback, which you told me not to do. Passing the src in New would lead to race conditions since the ship is set in this code. What I could do is add the component to the attack data instead and just daisy chain up the list to get the faction reference?